src/middleware/AdaptiveCardSubmitAction.ts

```diff
@@ -0,0 +1,24 @@
+import * as builder from "botbuilder";
+import { DialogMatches } from "../utils/DialogMatches";
+
+// Set the text field on message events to "adaptive card", if request is from an adaptive card
+export class AdaptiveCardSubmitAction implements builder.IMiddlewareMap {
+
+    public readonly receive = (event: builder.IEvent, next: Function): void => {
+        let currEvent = (event as any);
+
+        // if event text is blank, replyToId is not null and if defined key in adaptive card data is present in incoming payload
+        // to check if this request is from an adaptive card then set event text "adaptive card" to trigger Adaptive Card dialog,
+        // it's a work around that will be cleaned up later
+        if (currEvent.text === "" && currEvent.replyToId && currEvent.value)
```

Comment:
Let's clean up the comment so that it reflects the proposed changes below.

Have the check be:
if currEvent.text === "" && currEvent.replyToId && currEvent.value && currEvent.value.isFromAdaptiveCard && currEvent.value.messageText

In the block, rather than setting the currEvent.text to a constant set in DialogMatches, let's replace it with the 'messageText' value that was set in the value of the payload:
currEvent.text = currEvent.value.messageText

Reply:
Done.